src/box/vy_lsm.c

```diff
@@ -905,13 +905,8 @@ vy_lsm_set(struct vy_lsm *lsm, struct vy_mem *mem,
 
 	assert(vy_stmt_is_refable(entry.stmt));
 	assert(*region_stmt == NULL || !vy_stmt_is_refable(*region_stmt));
-
-	/* Abort transaction if format was changed by DDL */
-	if (!vy_stmt_is_key(entry.stmt) &&
-	    format_id != tuple_format_id(mem->format)) {
-		diag_set(ClientError, ER_TRANSACTION_CONFLICT);
-		return -1;
-	}
+	assert(vy_stmt_is_key(entry.stmt) ||
+	       format_id == tuple_format_id(mem->format));
```

Comment:
Why not `assert(format_id == tuple_format_id(mem->format));` if we abort all writers on DDL?

TBO, I don't understand why we needed this `vy_stmt_is_key(entry.stmt)` check at all.

Reply:
I got it - keys have a different format so we need to check format of non-key statements only.

Reply:
In Vinyl a statement may be in the form of a key or a tuple. The key format is used for DELETE statements if there are no secondary indexes or secondary index DELETEs are deferred. The key format is shared across all LSM trees, see `vy_lsm->env->key_format`.